### ISSUE TITLE
torch.where gives weird output

### ISSUE THREAD

**Issue by vainaijr** [pytorch/pytorch#28709]
![Screenshot (54)](https://user-images.githubusercontent.com/43755000/67604533-c43dbe00-f7b6-11e9-9843-2869638aade5.png)

**Comment by zhangguanheng66** [comment]
I reproduce the error with the following deterministic test.

```
t1 = torch.LongTensor([3, 4, 9, 7, 6, 8, 5, 1, 0, 2])
t2 = torch.zeros(10)
torch.where(t1>5, t1, t2)
```
@nairbv

**Comment by nairbv** [comment]
I wasn't able to repro on master, this may have been fixed by:

https://github.com/pytorch/pytorch/pull/28253

**Comment by vainaijr** [comment]
it still gives weird output
![Screenshot (63)](https://user-images.githubusercontent.com/43755000/67730703-2b6aa500-fa39-11e9-8bc1-b63169f99670.png)

**Comment by nairbv** [comment]
Ah I thought this was an issue with how we computed stride-bytes that was fixed on master, not sure why I wasn't able to repro when I first tried.

It looks like this issue may actually have been introduced by PR #25402 (I'm able to reproduce at commit 9c5a8997737eb0049b57cb43e988707d1d87596b, which was before type promotion landed). It wasn't intentional to add promotion capability to this operation.

cc @ifedan
